src/ui/Log/LogBoxManager.tsx

```diff
@@ -80,11 +157,27 @@ export function LogBoxManager(): React.ReactElement {
     rerender();
   }
 
+  function openExternally(log: Log): void {
+    if (log.foreignWindow) return;
+    const relativePath = process.env.NODE_ENV === "development" ? "./log/index.html" : "./dist/log/index.html";
+    const url = new URL(relativePath, window.location.href);
+    log.foreignWindow = window.open(url, "_blank", "popup");
+    rerender();
+  }
+
   return (
     <>
-      {logs.map((log) => (
-        <LogWindow key={log.id} script={log.script} id={log.id} onClose={() => close(log.id)} />
-      ))}
+      {logs
+        .filter((log) => !log.foreignWindow)
```

Comment:
Your render cannot depend on mutable state or it will desync when that state is changed.

Reply:
The previous implementation also had this: logs itself was updated in place, then rerender was called. I can switch up the design, but there might have been a reason to do it manually?